detox/src/artifacts/screenshot/ScreenshotArtifactPlugin.js

```diff
@@ -12,6 +12,15 @@ class ScreenshotArtifactPlugin extends TwoSnapshotsPerTestPlugin {
     return this.api.preparePathForArtifact(`${name}.png`, testSummary);
   }
 
+  async onBeforeCleanup(e) {
+    if (this.context.isAppReady === false) {
+      this._hasFailingTests = true;
```

Comment:
Hacky way, I admit, but this is because I have a not flexible inheritance model with these `super.onBeforeCleanup`.

The idea is to make sure that the unresponsive state of the app is considered a failing test, **even if no test was run**. How come?

Because if the outer `beforeAll` fails, then we don't get no `detox.beforeEach()` and `detox.afterEach()`. No beforeEach-afterEach, ====> no tests. But the run is not successful. So this way I at least make sure that if user runs Detox with `--take-screenshots failing`, we'll still pull that `appNotReady.png` from the device.

If I don't put that flag, we'll just remove that screenshot from the device, destroying the evidence. :smile:

I think, this place can be improved, but can't currently wrap my head around it. Also, I think a better fix would require more heavy refactoring in ArtifactsManager, and I am afraid to touch more. The PR is already a few hundred lines of code.

Reply:
By the way, why `=== false`? Because `isAppReady` can be `undefined, false, true`. If we never ran app and we get inside detox.cleanup(), it is still a legit case. That's why I chose those three states.